File: archive/t/typescript/factorial.ts
let error_msg: string = "Usage: please input a non-negative integer"
let num_str: string = (process.argv.length == 3) ? process.argv[2] : ""
let num: number = parseInt(num_str);
if (isNaN(num) || num < 0) {
    console.log(error_msg);
    process.exit(1);
}
let factorial: number = 1;
for (let i = 1; i <= num; i++) {
    factorial *= i;
}
console.log(factorial);

File: archive/o/octave/factorial.m
function factorial()
    usage = 'Usage: please input a non-negative integer';
    arg_list = argv();
    nargin = length(arg_list);
    if nargin == 0
        disp(usage);
        return;
    end

    n = str2num(arg_list{1});
    if length(n) ~= 1 || mod(n, 1) ~= 0 || n < 0
        disp(usage);
        return;
    end

    x = 1;
    for i = 1:n
        x = x * i;
    end

    disp(x);
end
